app/models/pulitzer/post.rb

```diff
@@ -1,11 +1,12 @@
 module Pulitzer
   class Post < ActiveRecord::Base
     extend ::FriendlyId
+    has_many :versions
     belongs_to :post_type
-    has_many :content_elements, dependent: :destroy
-    has_many :post_tags, dependent: :destroy
     delegate :post_type_content_element_types, to: :post_type
+    delegate :content_elements, :post_tags, to: :preview_version
```

Comment:
Got it, for the public implementation will be easier doing just `post.content_elements`, sure I will change this.